atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -140,15 +147,75 @@
 import com.palantir.timestamp.TimestampService;
 
 @SuppressWarnings("checkstyle:all")
+@RunWith(Parameterized.class)
 public class SnapshotTransactionTest extends AtlasDbTestCase {
+    private static final String SYNC = "sync";
+    private static final String ASYNC = "async";
+    private final String name;
+    private final Function<Transaction, Transaction> transactionWrapper;
+    private final Map<String, ExpectationFactory<KeyValueService, Cell, Long, LockService, Expectations>>
+            expectationsMapping =
+            ImmutableMap.<String, ExpectationFactory<KeyValueService, Cell, Long, LockService, Expectations>>builder()
+                    .put(SYNC, SnapshotTransactionTest.this::syncGetExpectation)
+                    .put(ASYNC, SnapshotTransactionTest.this::asyncGetExpectation)
+                    .build();
+
+    @FunctionalInterface
+    interface ExpectationFactory<One, Two, Three, Four, Five> {
+        Five apply(One one, Two two, Three three, Four four) throws Exception;
+    }
+
+    private Expectations syncGetExpectation(
+            KeyValueService kvMock,
+            Cell cell,
+            long transactionTs,
+            LockService lockMock) {
+        try {
+            return new Expectations() {{
+                oneOf(kvMock).get(TABLE, ImmutableMap.of(cell, transactionTs));
+                will(throwException(new RuntimeException()));
+                never(lockMock).lockWithFullLockResponse(with(LockClient.ANONYMOUS), with(any(LockRequest.class)));
+            }};
+        } catch (Exception e) {
+            throw new RuntimeException(e.getCause());
+        }
+    }
+
+    private Expectations asyncGetExpectation(
+            KeyValueService kvMock,
+            Cell cell,
+            long transactionTs,
+            LockService lockMock) throws Exception {
+        return new Expectations() {{
+            oneOf(kvMock).getAsync(TABLE, ImmutableMap.of(cell, transactionTs));
+            will(returnValue(Futures.immediateFailedFuture(new RuntimeException())));
+            never(lockMock).lockWithFullLockResponse(with(LockClient.ANONYMOUS), with(any(LockRequest.class)));
+        }};
+    }
+
     private TransactionConfig transactionConfig;
 
-    protected final TimestampCache timestampCache = new DefaultTimestampCache(
+    private final TimestampCache timestampCache = new DefaultTimestampCache(
             metricsManager.getRegistry(), () -> AtlasDbConstants.DEFAULT_TIMESTAMP_CACHE_SIZE);
-    protected final ExecutorService getRangesExecutor = Executors.newFixedThreadPool(8);
-    protected final int defaultGetRangesConcurrency = 2;
-    protected final TransactionOutcomeMetrics transactionOutcomeMetrics
+    private final ExecutorService getRangesExecutor = Executors.newFixedThreadPool(8);
+    private final int defaultGetRangesConcurrency = 2;
+    private final TransactionOutcomeMetrics transactionOutcomeMetrics
```

Comment:
I believe this is fine for this test, but in general please be cautious when reducing the visibility of things, _even if IDEs claim they aren't used elsewhere_, as you'll find that many internal projects access things they probably shouldn't